### PR TITLE
 Add 'Configure Project for Device Development' command

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: resources/templates/x86/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "applicationName": "",
+  "buildCommand": "bash /work/compile_app.sh ./src",
+  "buildTarget": "/work/azure-iot-sdk-c/cmake/iot_application"
```
Comment (dooriya) [inline resources/templates/x86/config.json]: Is the project name hard-coded?

Comment (dooriya) [reply resources/templates/x86/config.json]: As I known, 'iot_application' is the default project name which can be changed by user

File: src/Models/IoTContainerizedProject.ts [hunk 2]
```diff
@@ -149,7 +153,10 @@ export class IoTContainerizedProject extends IoTWorkbenchProjectBase {
           `Internal Error. Could not find iot workbench project file.`);
     }
 
-    // Step 3: Open project
+
+    // Step 3: Configure project
```
Comment (dooriya) [inline src/Models/IoTContainerizedProject.ts]: Please mark TODO if it is not implemented.

File: src/ProjectEnvironmentConfiger.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import * as utils from './utils';
+
+import {TelemetryContext} from './telemetry';
+import {FileNames, ScaffoldType, PlatformType, TemplateTag} from './constants';
+import {FileUtility} from './FileUtility';
+import {ProjectTemplate, TemplatesType} from './Models/Interfaces/ProjectTemplate';
+import {RemoteExtension} from './Models/RemoteExtension';
+import * as UIUtility from './UIUtility';
+
+const impor = require('impor')(__dirname);
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: impor => Please user better naming here.

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ export class CodeGeneratorCore {
   async generateDeviceCodeStub(
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext): Promise<boolean> {
-    const notRemote = RemoteExtension.checkNotRemoteBeforeRunCommand(context);
+    const notRemote = RemoteExtension.checkLocalBeforeRunCommand(context);
```
Comment (dooriya) [inline src/DigitalTwin/CodeGeneratorCore.ts]: Please update notRemote -> isRemote, which is more straightforward.
And also update the logic in 'checkLocalBeforeRunCommand'

File: resources/templates/x86/config.json (hunk 1 above)
Comment (dilin-MS) [reply resources/templates/x86/config.json]: [TODO] Just a placeholder file here. Will modify the file for extensibility later.

File: src/Models/IoTContainerizedProject.ts (hunk 2 above)
Comment (dilin-MS) [reply src/Models/IoTContainerizedProject.ts]: Fixed.

File: src/ProjectEnvironmentConfiger.ts (hunk 3 above)
Comment (dilin-MS) [reply src/ProjectEnvironmentConfiger.ts]: It is the original name of the npm library.

File: src/DigitalTwin/CodeGeneratorCore.ts (hunk 4 above)
Comment (dilin-MS) [reply src/DigitalTwin/CodeGeneratorCore.ts]: Fixed.

**Pull request merged**